Version file introduced and version set to 0.1.0

erikzenker opened the pull request:
This PR prepares the version for the next release

BenjaminW3 commented on include/alpaka/version.hpp:
```diff
@@ -0,0 +1,26 @@
+
+#define ALPAKA_VERSION_MAJOR 0
+#define ALPAKA_VERSION_MINOR 1
+#define ALPAKA_VERSION_PATCH 0
```
Could you please add the lines you removed from alpaka.hpp? I find the `ALPAKA_VERSION` using the predef version macros really helpful. However, they should now use the major, minor and patch defines. You are missing the trailing newlin.

BenjaminW3 commented on alpakaConfig.cmake:
```diff
@@ -527,17 +527,20 @@ ENDIF()
 # Set return values.
 #-------------------------------------------------------------------------------
-SET(alpaka_VERSION "${_ALPAKA_VER_MAJOR}.${_ALPAKA_VER_MINOR}.${_ALPAKA_VER_PATCH}")
+set(alpaka_VERSION "${ALPAKA_VERSION_MAJOR}.${ALPAKA_VERSION_MINOR}.${ALPAKA_VERSION_PATCH}")
```
Could you please write the cmake functions uppercase as long as the other lines are also uppercase?

BenjaminW3 commented:
You will have to add the include of the version.hpp file into the alpaka.hpp file.

erikzenker commented:
* Added the boost version number again. It is using now the major, minor, patch defines
* Added newline at the end of the version.hpp file
* Corrected cmake function the uppercase writing
